dashboard/app/controllers/api/v1/projects/personal_projects_controller.rb

```diff
@@ -0,0 +1,12 @@
+class Api::V1::Projects::PersonalProjectsController < ApplicationController
+  # GET /api/v1/projects/personal/
+  def index
+    unless current_user
+      redirect_to "/", flash: {alert: 'You must be logged in to access your projects'}
```

Comment:
The redirect and flash is a weird behavior for a JSON api.  It might make more sense to
```rb
return head :forbidden unless current_user
```
Or use one of the CanCanCan patterns for authorizing this route (@aoby help me out here?)

Reply:
+1, API routes should not redirect and or display a flash message. The one exception we have is that some of them redirect to the sign in page when not signed in. This is a result of default CanCan authorize behavior, not something we have written code to do explicitly. Usually these would only be called from a UI that is only displayed to logged in users anyway, so that path should not be hit. If it's part of the normal expected flow, APIs should not redirect.

re: authorization
 1. We can still use CanCan(Can) here at least to authorize, even if the "resource" is not a Rails model. CanCan resources are ideally models, but can really be any arbitrary symbol. Here's an [example](https://github.com/code-dot-org/code-dot-org/blob/fbcc1e45ed56089c4dc95c85e8b32c11e5ccf409/dashboard/app/models/ability.rb#L97). However Dave is right we can't use it to load the filtered set because it's not a Rails model. Or more specifically it's not a class that implements [accessible_by](https://www.rubydoc.info/github/CanCanCommunity/cancancan/CanCan%2FModelAdditions%2FClassMethods:accessible_by).
 2. I like the idea of creating a wrapper model class which is backed by `storage_apps` with an association from `User`. In fact, we can get the `load_and_authorize` stuff to work even without that association by using a class (ActiveRecord or not) as our resource which implements `accessible_by` to do the filtering. See https://github.com/CanCanCommunity/cancancan/wiki/Fetching-Records